intersections/build.boot

```diff
@@ -0,0 +1,113 @@
+       :url         "http://www.quazistax.com/testIntersection.html"
+       :scm         {:url "git://github.com/thelonious/kld-intersections.git"}
+       :license     {"BSD" "http://opensource.org/licenses/BSD-3-Clause"}})
+
+(require '[cljsjs.boot-cljsjs.packaging :refer :all]
+         '[nha.boot-uglify.minify-js :as minify-js]
+         '[boot.core :as c]
+         '[boot.util :as u]
+         '[boot.file :as file]
+         '[clojure.java.io :as io])
+
+(defn normalize-path
+  "Adds a / if missing at the end of the path."
+  [path]
+  (if-not (= \/ (last path))
+    (str path "/")
+    path))
+
+(c/deftask uglify
+  "Create the minified file
+
+  This task also copies the minified file in the development folder."
+  [_ working-dir DIR str "Path where the js files to uglify will be in"
+   _ file-name NAME [str] "Vector of JS source file names to uglify (order counts)"
+   _ prod-dir DIR str "The package production dir"
+   _ dev-dir DIR str "The package development dir"
+   _ prod-name NAME str "File name for the minified file"
+   _ dev-name NAME str "File name for the development file"]
+  (let [tmp (c/tmp-dir!)]
+    (c/with-pre-wrap fileset
+      (let [real-dir (->> (c/ls fileset)
+                          (c/by-re [(re-pattern working-dir)])
+                          first
+                          c/tmp-file
+                          .getParent
+                          normalize-path)
+            prod-file (io/file tmp (str prod-dir prod-name))
+            dev-file (io/file tmp (str dev-dir dev-name))]
+        (u/dbug* "Minifying content of: %s\n" real-dir)
+        (let [uglify-result (minify-js/minify-js (mapv #(str real-dir %) file-name)
+                                                 prod-file)]
+          (u/dbug* "Uglify result: %s\n" (u/pp-str uglify-result)))
+        (u/dbug* "Minified file: %s\n" prod-file)
+        (u/dbug* "Development file: %s\n" dev-file)
+        (io/make-parents dev-file)
+        (file/copy-atomically prod-file dev-file)
+        (-> fileset (c/add-resource tmp) c/commit!)))))
+
+(deftask package []
+  (let [uglify-dir (normalize-path "uglify/")
+        prod-dir (normalize-path "cljsjs/intersections/production/")
+        dev-dir (normalize-path "cljsjs/intersections/development/")]
+    (comp
+     (download :url "https://raw.githubusercontent.com/thelonious/kld-intersections/development/lib/Intersection.js"
+               :checksum "2FF91BB218A7318DA4F7625A8DCFD27C")
+     (download :url "https://raw.githubusercontent.com/thelonious/kld-intersections/development/lib/IntersectionParams.js"
+               :checksum "6C7979D5AD687F729138DFDE61BD3330")
+     (sift :move {#"^Intersection.js" (str uglify-dir "Intersection.js")})
+     (sift :move {#"^IntersectionParams.js" (str uglify-dir "IntersectionParams.js")})
+
+     ;; dep kld-affine
+     (download :url "https://raw.githubusercontent.com/thelonious/kld-affine/development/lib/Point2D.js"
```

Comment:
This is fair, but I haven't got the time for splitting it and as I say below, it was the only way to make it working...